src/analyzer.rs

```diff
@@ -291,6 +298,16 @@ mod test {
         );
     }
 
+    #[test]
+    fn test_japanese_language() {
+        let analyzer = Analyzer::new(AnalyzerConfig::<Vec<u8>>::default());
+
+        let orig = "関西国際空港限定トートバッグ";
```

Comment:
@ManyTheFish 
Seems this is issue about analyzer.
If the string looks like following, seems it is selecting `jieba` tokenizer, because first character is `関`, and this is both being used in chinese/japanese.

How could this be solved?

Reply:
Hey! @djKooks! 

In the [default function of the AnalizerConfig](https://github.com/meilisearch/tokenizer/blob/c2399c3f879144ad92e20ae057e14984dfd22781/src/analyzer.rs#L127-L141)  you should add a new pipeline in order to use your new tokenizer when a Japanese script is detected:

```rust
    fn default() -> Self {
        let mut pipeline_map: HashMap<(Script, Language), Pipeline> = HashMap::new();


        // Latin script specialized pipeline
        pipeline_map.insert((Script::Latin, Language::Other), Pipeline::default()
            .set_tokenizer(LegacyMeilisearch));

        // Chinese script specialized pipeline
        pipeline_map.insert((Script::Mandarin, Language::Other), Pipeline::default()
            .set_pre_processor(ChineseTranslationPreProcessor)
            .set_tokenizer(Jieba::default()));
            
        // Japanese script specialized pipeline
        pipeline_map.insert((Script::Katakana, Language::Other), Pipeline::default()
            .set_tokenizer(LinderaTokenizer::new(Mode::Normal, "/path/to/ja-dic")));
        pipeline_map.insert((Script::Hiragana, Language::Other), Pipeline::default()
            .set_tokenizer(LinderaTokenizer::new(Mode::Normal, "/path/to/ja-dic")));

        // Korean script specialized pipeline
        pipeline_map.insert((Script::Hangul, Language::Other), Pipeline::default()
            .set_tokenizer(LinderaTokenizer::new(Mode::Normal, "/path/to/ko-dic")));

        AnalyzerConfig { pipeline_map, stop_words: None }
    }
 ```
 
 I don't know Japanese or Korean but the previous lines might help you to fix your test 👍 
 
 Don't hesitate to ping me if you need more help!

Reply:
@ManyTheFish thanks for your feedback. I have 2 question.
- I've make few change in following commit, to setup dictionary for tokenizer. But because I'm not quite good in Rust, I think you can suggest more better way to define Lindera tokenizer which extends current base tokenizer. Could you give me some suggestion?
- I've tested  the code you've shared above. Current main problem is that if first character of text are the one which are being used commonly in CJK(Chinese/Japanese/Korean), it will indicate text as Chinese because ChineseTokenizer comes first(maybe I can be wrong). Is there some solutions about when there are multiple languages in text?

Reply:
Hello @djkooks, 😄 

for your first point, I wrote so comments
about your second point, I agree with you about the fact that the script detection is not the best 😅; I will enhance it sooner or later 😊